Display issue type when creating new issue

nickymarino opened the pull request:
### Description of Changes:

The header for creating a new issue now says "Report Train Issue", "Report Bus Issue", or "Report an Issue" depending on the line type

### How This Was Tested:

- I tested `/issues/new?line_type=train`, `/issues/new?line_type=bus`, and `/issues/new` (a nil line type)
- A new UI test named `visiting the index` in `test/system/issue_creations_test.rb` was added

### Related Issue(s) or Specifications:
- None

### Checklist:
- [X] Code follows code style of this project
- [X] Tests added to cover changes
- [X] All new and existing tests passing

### Questions / Additional Notes:

None

vkoves commented:
@nickymarino - I have confirmed that the routes you tested work fine and update the header, but should links somewhere be updated, or is this just a future facing change? If I report an issue on a stop for instance, it does `/issues/new?line_id=9&stop_id=9`, which doesn't take advantage of this new functionality.

nickymarino commented:
@vkoves I'm not sure whether links should be updated. I worked with Tomer during one of the Tuesday events to change the headers for train/bus issues based on the wireframes we have, and I wasn't aware `/issues/new?line_id=9&stop_id=9` was a possible link

I can definitely update either the link methods or the header update depending on which way we should go though

nickymarino commented:
To keep everyone in the loop, I’ll work on making the above UI test pass

nickymarino commented:
Implemented the logic to the issue `new.html.erb` to pass the UI test

The UI test itself was breaking, and I'm still not quite sure why. 

Capybara was able to find the issue creation link using `find("#create-issue-#{stops(:fullerton).id}")`, but the `.click` event wasn't working correctly for some reason. Other options like `click_link` and `click_on` didn't work either.

As a workaround, I changed it to visit the tag's href instead of trying to click it:

`visit find("#create-issue-#{stops(:fullerton).id}")['href']`

vkoves commented on app/views/issues/new.html.erb:
```diff
@@ -1,11 +1,12 @@
 <% if @line_type %>
 	<h1>Report <%= @line_type.capitalize %> Issue</h1>
+<% elsif @line_id %>
+	<% @line=Line.find_by_onestop_id(@line_id) %>
```
Logic should be kept in the controller as much as possible, especially anything that does a DB query like this. I'd recommend moving it into your controller, and then in the view you can just do `<% elsif @line %>`

vkoves commented on app/controllers/issues_controller.rb:
```diff
@@ -26,6 +26,8 @@ def new
 
     @line_type = params[:line_type]
+    @line_id = params[:line_id]
+    @stop_id = params[:stop_id]
```
You added this instance variable, but I don't see it used anywhere. Am I missing something?

vkoves commented on test/fixtures/lines_stops.yml:
```diff
@@ -2,3 +2,6 @@ one:
 
+fullerton_red_line:
+  stop_onestop_id: "fullerton-id"
+  line_onestop_id: "red-line"
```
Minor thing, but can you add a new line to the end of this file and `stops.yml`? It's a good general practice I like to follow, as not doing it can cause problems in some editors

vkoves commented on test/system/issue_creations_test.rb:
```diff
@@ -0,0 +1,14 @@
+require "application_system_test_case"
+
+class IssueCreationsTest < ApplicationSystemTestCase
+  test "visiting the index" do
```
This test description doesn't describe what the test is looking for - my rule of thumb is to ask, if this test failed would the name tell you what was wrong? In this case, I'd suggest changing it to something like "Creating an issue on a train station is titled 'Report Train Issue'" or something like that.

vkoves commented on test/test_helper.rb:
```diff
@@ -25,4 +25,13 @@ def sign_in(user:, password:)
       "user[password]" => password
   end
+
+  def sign_in_ui(user_fixture, password:)
```
Can you document this function? When should we use this versus `sign_in` above?

vkoves commented:
@nickymarino - feel free to merge in if you are ready, now that I've approved. I forgot to mention that in my approval :grin:

vkoves commented:
I realized you might not have access to merge, so merging this in :sweat_smile:

nickymarino commented:
@vkoves Thanks! I'm not sure if I have access to merge
